pkg/backoff/backoff.go

```diff
@@ -73,6 +91,30 @@ func (r *retry) WaitNext(ctx context.Context) bool {
 	}
 }
 
```

Comment:
nit: it will be helpful to leave a comment like:
```
// Do executes the given operation for the specified number of times. If an error occurs for which doesn't allow retrying, generate a new error with `NewError` and return it within the operation function.
```

Reply:
Sure. Add a comment.